src/AspNet.Security.OAuth.Yahoo/YahooAuthenticationHandler.cs

```diff
@@ -0,0 +1,92 @@
+    public class YahooAuthenticationHandler : OAuthAuthenticationHandler<YahooAuthenticationOptions> {
+        public YahooAuthenticationHandler([NotNull] HttpClient client)
+            : base(client) {}
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens) {
+            var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint);
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", tokens.AccessToken);
+
+            var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+            response.EnsureSuccessStatusCode();
+
+            var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
+
+            // Extract the profile object from the result payload
+            var profile = payload.Value<JObject>("profile");
+
+            // Extract all the claims from the profile
+            if (profile != null) {
+                identity.AddOptionalClaim(ClaimTypes.NameIdentifier, YahooAuthenticationHelper.GetIdentifier(profile), Options.ClaimsIssuer)
+                    .AddOptionalClaim(ClaimTypes.Name, YahooAuthenticationHelper.GetNickname(profile), Options.ClaimsIssuer)
+                    .AddOptionalClaim("urn:yahoo:familyname", YahooAuthenticationHelper.GetFamilyName(profile), Options.ClaimsIssuer)
+                    .AddOptionalClaim("urn:yahoo:givenname", YahooAuthenticationHelper.GetGivenName(profile), Options.ClaimsIssuer)
+                    .AddOptionalClaim("urn:yahoo:profile", YahooAuthenticationHelper.GetProfileUrl(profile), Options.ClaimsIssuer)
+                    .AddOptionalClaim("urn:yahoo:profileimage", YahooAuthenticationHelper.GetProfileImageUrl(profile), Options.ClaimsIssuer);
+            }
+
+            var notification = new OAuthAuthenticatedContext(Context, Options, Backchannel, tokens, payload) {
+                Principal = new ClaimsPrincipal(identity),
+                Properties = properties
+            };
+
+            await Options.Notifications.Authenticated(notification);
+
+            if (notification.Principal?.Identity == null) {
+                return null;
+            }
+
+            return new AuthenticationTicket(
+                notification.Principal, notification.Properties,
+                notification.Options.AuthenticationScheme);
+        }
+
+        protected override async Task<OAuthTokenResponse> ExchangeCodeAsync(string code, string redirectUri) {
+            var tokenRequestParameters = new Dictionary<string, string>() {
+                {"client_id", Options.ClientId},
+                {"redirect_uri", redirectUri},
+                {"client_secret", Options.ClientSecret},
```

Comment:
Sounds like `client_id` and `client secret` are flowed both in the request form and in the `Authorization` header. Do you need both methods?